src/Storages/MergeTree/MergeTreeBackgroundExecutor.h

```diff
@@ -225,10 +327,8 @@ class MergeTreeBackgroundExecutor final : boost::noncopyable
     Poco::Logger * log = &Poco::Logger::get("MergeTreeBackgroundExecutor");
 };
 
-extern template class MergeTreeBackgroundExecutor<MergeMutateRuntimeQueue>;
-extern template class MergeTreeBackgroundExecutor<OrdinaryRuntimeQueue>;
-
-using MergeMutateBackgroundExecutor = MergeTreeBackgroundExecutor<MergeMutateRuntimeQueue>;
-using OrdinaryBackgroundExecutor = MergeTreeBackgroundExecutor<OrdinaryRuntimeQueue>;
+extern template class MergeTreeBackgroundExecutor<RoundRobinRuntimeQueue>;
+extern template class MergeTreeBackgroundExecutor<PriorityRuntimeQueue>;
```

Comment:
Why does it compile? Line 279 requires `pending.updatePolicy(policy);` where `pending` has type `Queue` (template arg of `MergeTreeBackgroundExecutor`). But `RoundRobinRuntimeQueue` and `PriorityRuntimeQueue` do not have this method.

Reply:
Well, seems like clang-14 (I'm still using it locally) does not completely support concepts. Ok, so it's time to upgrade :)

Reply:
All this complications are due to the use of explicit template instantiations. Without it compiler would not even try to instantiate methods that are not called. And concepts (or SFINAE) would not be needed here.